roles/edpm_ovn/tasks/configure.yml

```diff
@@ -14,6 +14,33 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+- name: Apply ovsdb configuration from configmaps
+  block:
+  - name: discover configmap files in {{ edpm_ovn_config_src }}
+    ansible.builtin.find:
+      paths: "{{ edpm_ovn_config_src }}"
+      file_type: file
+      recurse: yes
+    register: edpm_ovn_configmaps
+    changed_when: false
+    check_mode: false
+    delegate_to: localhost
+
+  - name: read configmap file contents
+    ansible.builtin.slurp:
+      src: "{{ item.path }}"
+    loop: "{{ edpm_ovn_configmaps.files }}"
+    register: ovn_configmap_contents
+    delegate_to: localhost
+    when: edpm_ovn_configmaps.matched > 0
+
+  - name: apply configuration to Open_vSwitch:external_ids
+    ansible.builtin.shell: |
+      {% for line in item.content | b64decode | split('\n') %}
+      ovs-vsctl set open . external-ids:{{ line }}
```

Comment:
As we are in early phase thinking more on this wouldn't be better to write contents in dictionary format in config map and include_vars: <file> and do set these like line 70?

Reply:
I don't want ovn-operator to assume ansible format and variable names. The fact the ConfigMap is consumed by EDPM-Ansible seems to me incidental and is not part of the ovn-operator definition. And most other operators use some form of "property" (INI) file for ConfigMaps, so key=value seemed fitting. (Of course, the final format is not too important because either way it's not native to OVS - since it doesn't have a concept of configuration files.)

That said, we could still make it a "property" (yaml) file that is loaded as `include_vars`. (As long as names in ConfigMaps are not prepended with `edpm` prefixes.) What does it buy us?

Reply:
No, just to be clear i didn't meant to use edpm variable names in ConfigMap, what i meant was to have a generated file with contents something like below in a file like ovsdb-config.yaml, load it with include_vars and then merge the map "ovs_external_ids_override" with "edpm_ovn_ovs_external_ids" and let line 70 apply it:- 
ovsdb_external_ids_override:
  ovn-remote: "<SB DB endpoint>"

Reply:
@karelyatin ah so the goal would be to replace the step that parses the key=value file with (perhaps somewhat simpler) step that loads the yaml file as ansible variables + merges it into the edpm_ovn_ovs_external_ids?

I don't know if it saves much of complexity (I have little experience with ansible), but I can try. To confirm, you see the lines 39-41 as too complex?

Reply:
On second thought, it's not too complex, and I don't think that we should adjust API (ConfigMap) to a tool in question (ansible), so the fact that a different format could be *marginally* simpler for ansible to load shouldn't drive the decision either way. Considering that respinning will take some of dev time and CI cycles, unless there's a good reason to go there, I prefer to leave it as-is.

Reply:
yeap right it was not about complexity.  If edpm cores are ok with current implementation i can live with it :)

Reply:
I switched to facts, so it's now yaml.